DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/svd_testcase.hpp

```diff
@@ -0,0 +1,222 @@
+#ifndef __SVD_TESTCASE__
+#define __SVD_TESTCASE__
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+#include "golden_pca.hpp"
+#include "print_matrix.hpp"
+#include "svd.hpp"
+#include "svd_testbench_tool.hpp"
+
+// test case class to create test cases
+// Instantiate test case like this:
+// SVDTestcase<typeT, rows, cols> new_testcase(
+//     std::vector<std::vector<float>>{ ... }
+//     std::vector<float>{ ... });
+template <typename T, unsigned rows_A, unsigned cols_A, unsigned matrix_count>
+struct SVDTestcase {
```

Comment:
I like this class for running test cases, but please clarify in the comments for each function what it's for (i.e. for generating input or checking output?) there are a lot of functions here (e.g. `ExtractSingularValue()` and it's not clear what they're for.

I'm not clear why you have a separate file called `svd_testbench_tool.hpp` with what appear to be more utility functions. What is the requirement for a function to show up in here vs `svd_testbench_tool.hpp`?